webrender_traits/src/webgl.rs

```diff
@@ -973,6 +994,16 @@ impl WebGLCommand {
         chan.send(shader).unwrap();
     }
 
+    fn create_vertex_array(gl: &gl::Gl, chan: MsgSender<Option<WebGLVertexArrayId>>) {
+        let vao = gl.gen_vertex_arrays(1)[0];
+        let vao = if vao == 0 {
```

Comment:
That doesn't sound like a case where we just check for the returned ID then.
> I think that it can return 0 in situations where the GLContext is not correctly set

Shouldn't it be forbidden to do any GL calls in this case?

> maybe some GPUs have some limits for the quantity of VAOs that can be created

That would be a GL error then. What is our WebGL implementation policy with regards to GL errors?

> A similar check is done in other crate calls like create_framebuffer, create_texture.

Thanks for pointing this out. I'd like to investigate further.